### ISSUE TITLE
Feature request: Data Source for : autoscaling_group

### ISSUE THREAD

**Issue by toredash** [terraform-providers/terraform-provider-aws#6758]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Feature request to add autoscaling_group as a data source. There is this (https://www.terraform.io/docs/providers/aws/d/autoscaling_groups.html) but that does not export information I'm after, e.g. min_size.

Data source could take `arn` as input. API already provides this: https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_DescribeAutoScalingGroups.html

It should support all of the attributes currently provided when creating an autoscaling_group:
```
id - The autoscaling group id.
arn - The ARN for this AutoScaling Group
availability_zones - The availability zones of the autoscale group.
min_size - The minimum size of the autoscale group
max_size - The maximum size of the autoscale group
default_cooldown - Time between a scaling activity and the succeeding scaling activity.
name - The name of the autoscale group
health_check_grace_period - Time after instance comes into service before checking health.
health_check_type - "EC2" or "ELB". Controls how health checking is done.
desired_capacity -The number of Amazon EC2 instances that should be running in the group.
launch_configuration - The launch configuration of the autoscale group
vpc_zone_identifier (Optional) - The VPC zone identifier
load_balancers (Optional) The load balancer names associated with the autoscaling group.
target_group_arns (Optional) list of Target Group ARNs that apply to this AutoScaling Group
```

### New or Affected Resource(s)

* aws_XXXXX

### Potential Terraform Configuration

```hcl
data "aws_autoscaling_group" "foobar" {
  arn = "...."
}

```

### References
This is _not_ to be confused with aws_autoscaling_group**s**: https://www.terraform.io/docs/providers/aws/d/autoscaling_groups.html

**Comment by bflad** [comment]
Support for a new `aws_autoscaling_group` (non-plural) data source has been merged and will release with version 1.54.0 of the AWS provider, likely early January due to end of year break. 👍

**Comment by bflad** [comment]
This has been released in [version 1.54.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1540-december-21-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
